src/decide.ts

```diff
@@ -143,7 +140,9 @@ export function compareFileTreesPartial(
 			matchedSizes += candidateFile.length;
 		}
 	}
-	return matchedSizes / candidate.length >= 1 - fuzzySizeThreshold;
+	const totalPieces = Math.ceil(candidate.length / candidate.pieceLength);
+	const missingPieces = Math.ceil(matchedSizes / candidate.pieceLength);
+	return missingPieces / totalPieces <= fuzzySizeThreshold;
```

Comment:
I like this as it's more accurate to "how much am I willing to download for a partial match"